substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -3326,6 +3371,12 @@ impl<T: Config> Pallet<T> {
 		member_account: T::AccountId,
 		extra: BondExtra<BalanceOf<T>>,
 	) -> DispatchResult {
+		// ensure member is not in an un-migrated state.
+		ensure!(
```

Comment:
I think it would be cleaner to either do this check at the extrinsic level or at the lower level implementations (i.e. here). I personally prefer to keep it at the extrinsic level rather than here. Or is there any other reason for placing this check here instead of in `fn bond_extra`?

Reply:
Did because bond_extra is getting called from two places and all business logic is inside. But to keep the convention, will move checks to top level fn.